model/common/src/icon4py/model/common/grid/geometry.py

```diff
@@ -178,39 +175,108 @@ def __init__(
         self.register_provider(input_fields_provider)
         self._register_computed_fields()
 
+    @staticmethod
+    def with_geometry_type(
+        grid: icon.IconGrid,
+        decomposition_info: decomposition.DecompositionInfo,
+        backend: gtx_typing.Backend | None,
+        coordinates: gm.CoordinateDict,
+        extra_fields: gm.GeometryDict,
+        metadata: dict[str, model.FieldMetaData],
+    ) -> "GridGeometry":
+        return GridGeometry(
+            grid,
+            decomposition_info,
+            backend,
+            coordinates,
+            extra_fields,
+            metadata,
+        )
+
```

Comment:
This was only needed for dispatching to the different subclass constructors, but can be removed now:
```suggestion
```